packages/eas-cli/src/devices/actions/create/currentMachineMethod.ts

```diff
@@ -0,0 +1,132 @@
+import spawnAsync from '@expo/spawn-async';
+import chalk from 'chalk';
+import os from 'os';
+
+import { ExpoGraphqlClient } from '../../../commandUtils/context/contextUtils/createGraphqlClient';
+import { AppleDeviceMutation } from '../../../credentials/ios/api/graphql/mutations/AppleDeviceMutation';
+import { AppleTeam } from '../../../graphql/generated';
+import Log from '../../../log';
+import { ora } from '../../../ora';
+import { confirmAsync } from '../../../prompts';
+import { DeviceCreateError } from '../../utils/errors';
+import {
+  DeviceData,
+  printDeviceData,
+  promptForDeviceClassAsync,
+  promptForNameAsync,
+} from './utils';
+
+export async function runCurrentMachineMethodAsync(
+  graphqlClient: ExpoGraphqlClient,
+  accountId: string,
+  appleTeam: Pick<AppleTeam, 'appleTeamIdentifier' | 'appleTeamName' | 'id'>
+): Promise<void> {
+  Log.newLine();
+  Log.log(chalk.white('Checking if current machine is an Apple Silicon one.'));
+  if (!isMachineAppleSilicon()) {
+    throw new DeviceCreateError(
+      "Current machine is not of Apple Silicon type - provisioning UDID can't be added automatically."
+    );
+  }
+  Log.log(chalk.green('Check successful.'));
+
+  await collectDataAndRegisterDeviceAsync(graphqlClient, { accountId, appleTeam });
+}
+
+function isMachineAppleSilicon(): boolean {
+  return os.cpus()[0].model.includes('Apple M') || os.arch() === 'arm64';
```

Comment:
Theoretically nothing, but since a) I don't have access to various machines to test, and b) I'm not sure the naming conventions/function definitions won't change in the future; I thought it would be better safe than sorry and check for either of these

Reply:
I think other OS's can probably be arm64 so this will have a lot of false positives as written: https://en.wikipedia.org/wiki/Category:ARM_operating_systems

Reply:
You can also add a check for `process.platform === 'darwin'`

Reply:
I think checking for `Apple M` is correct for now, and we can alter the check when it changes (not try to predict it too much)